### ISSUE TITLE
Blender 2.81 upgrade

### ISSUE THREAD

**Issue by animate1978** [animate1978/MB-Lab#164]
Now that Blender 2.81 is in Beta I think it is time to look at it to upgrade MB-Lab from the current version minimum of Blender 2.80.74.
https://builder.blender.org/download/ 

_Here are some of the highlights from this upcoming version._

**Cycles**

- New White Noise Texture node, producing a random number based on input values.
- Noise Texture node: support 1D, 2D, and 4D noise.
- Musgrave Texture node: support 1D, 2D, and 4D musgrave.
- Voronoi Texture node: support 1D, 2D, and 4D voronoi. Add more feature types. 
- New Volume Info node, for convenient access to Color, Density, Flame, and Temperature attributes of smoke domains.
- Object Info node: new Object Color output.
- New Map Range node. Linearly remap a value from one range to another.
- New Clamp node. Clamps a value between a maximum and a minimum values.
- Math node: shows one or two inputs depending on how many are needed by the operation.
- Vector Math node: new operations, and show number of inputs depending on the operation.
- Mapping node: location, rotation and scale are now node inputs that can be linked to other nodes.
- New Vertex Color node, for convenient access to Vertex Colors and their alpha channel.

**So the above could potentially change the shaders in future versions. I have not played around with these new features and changes yet as I have been working on MB-Lab 1.7.6 in Blender 2.80 so I can't say for certain what will change in the shaders for MB-Lab.**

**EEVEE**
**Shadows**
 
Soft shadows were changed and now enabled with the Use Soft Shadow  option in the render properties Shadow panel. This will progressive  smooth shadows, in combination with a more conventional shadow filtering  method (PCF). Exponential and Variance Shadow Mapping methods were  removed.
This improves shadow accuracy, makes light and shadow setup simpler, and improves performance for soft shadows. 
- Enabling the Use Soft Shadow option will now anti-alias the shadow-maps (even if the light size is 0).
- An automatic bias is computed based on the shadow map resolution.  Increasing the shadow resolution will reduce the bias and thus light  leaking.
- Reducing the shadow-map resolution will incidentally increase the  filter size (previously done by increasing the light shadow's Softness) and will lead to less shadow banding. However, this is a global parameter and cannot be tweaked per light.
- The shadow Bias parameter now adds an additional bias to avoid false shadowing on depth discontinuity.

**Other notable improvements:** 
- Contact shadows are now following the light shape. The contact shadow Softness has been removed.
- Sun lights now have their clip distances automatically computed.
- Instanced lights and shadow casters are now supported.

**Transparent BSDF**
Transparency handling has been reworked and now supports the same BSDF combinations as Cycles.
 
**Associated node tree**   
The Additive and Multiply blend mode have been replaced by shader  equivalent that is compatible with Cycles. Compatibility code has been  added to load blend file containing materials using these deprecated  blend modes, but there is a corner case where it can fail. 
 
 Node setup replacing deprecated blend modes. 
Holdout
Holdout is now supported and works with both opaque and semi transparent surfaces.
 

**Other Changes**

- Bump mapping was changed to give more accurate results closer  to Cycles. In particular it makes bump mapping with procedural textures  work much better than before, at the cost of some performance.
- The Separate Albedo option is now always on and removed from the UI. Mixing (the active) SSS node and other BSDFs is now more correct.

**The changes that really apply to MB-Lab are the Transparency BSDF and the Bump Mapping.**

**Python Upgrade**
Python has been upgraded from **3.7.0** to version, **3.7.4.** 
API Changes

- IDMaterials.clear() and IDMaterials.pop() no longer have an update_data parameter. It now always behave as if update_data=True was passed.
- Bone.use_inherit_scale and EditBone.use_inherit_scale are deprecated in favor of the new inherit_scale enum property.
- CopyRotationConstraint.use_offset is deprecated in favor of the new mix_mode enum.
- CyclesShaderWrapper has been removed developers should be using PrincipledBSDFWrapper now.
- ShaderNodeMapping was rewritten to have dynamic inputs, so direct access to translation, rotation and scale has been removed and is now done via inputs['Location'], inputs['Rotation'] and inputs['Scale']. Also min/max was removed entirely and should be replaced by access to the Extension parameter of the Texture node
- Bone properties related to its position and  orientation have been marked read only, because modifying them was never  expected by the code and resulted in inconsistent state. The rest pose  should only be modified through EditBone.

**I am going to have to go through the code and see if any of the above needs to be changed.**

**Comment by animate1978** [comment]
As of this morning I get this error

Switching to fully guarded memory allocator.
Blender 2.81 (sub 15)
Build: 2019-10-12 23:01:11 Linux Release
argv[0] = /home/farishta/blender-2.81/blender
argv[1] = --debug
Read prefs: /home/farishta/.config/blender/2.81/config/userpref.blend
read file /home/farishta/.config/blender/2.81/config/startup.blend
  Version 280 sub 75 date 2019-07-29 14:47 hash f6cb5f54494e
found bundled python: /home/farishta/blender-2.81/2.81/python
Traceback (most recent call last):
  File "/home/farishta/blender-2.81/2.81/scripts/modules/addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "/home/farishta/.config/blender/2.81/scripts/addons/MB-Lab/__init__.py", line 38, in <module>
    from . import humanoid, animationengine, proxyengine
  File "/home/farishta/.config/blender/2.81/scripts/addons/MB-Lab/humanoid.py", line 34, in <module>
    from . import morphengine, skeletonengine, algorithms, proxyengine, materialengine
  File "/home/farishta/.config/blender/2.81/scripts/addons/MB-Lab/materialengine.py", line 26, in <module>
    import numpy as np
  File "/home/farishta/blender-2.81/2.81/python/lib/python3.7/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/home/farishta/blender-2.81/2.81/python/lib/python3.7/site-packages/numpy/core/__init__.py", line 67, in <module>
    raise ImportError(msg.format(path))
ImportError: Something is wrong with the numpy installation. While importing we detected an older version of numpy in ['/home/farishta/blender-2.81/2.81/python/lib/python3.7/site-packages/numpy']. One method of fixing this is to repeatedly uninstall numpy until none is found, then reinstall this version.

Looks like something with numpy, going to look into it

**Comment by animate1978** [comment]
I took the site-package of numpy from Blender 2.80 and replaced the site-package in 2.81, this cleared up the issue, going to check on the Blender bug reports to see if this is an issue yet.

**Comment by animate1978** [comment]
https://developer.blender.org/T68833

This is not a bug, just delete everything in your Blender directory, then extract Blender 2.81 beta, should clear up.

**Comment by Hopefullyidontgetbanned** [comment]
blender 2.81 should release in the next few days, likely tomorrow from their official estimates.

**Comment by animate1978** [comment]
Yes I've been awaiting this release, however not sure if I want to bump up the min version MB-Lab will run only because there has not been really anything changed that can't run on the 2.80 release version. I will think it over the next couple of days.

**Comment by animate1978** [comment]
Something changed between versions, obviously, it does in fact affect the iris shader (somehow, not sure at this point what).

What happens is that if a user makes a MB-Lab character in 2.80, saves it in 2.81 and then tries to open it up in 2.80 again.... well the below screenshots give an example of what happens. It is not advisable to open up files of latest versions in previous versions of Blender, there should not be much change but there is enough.

![iris_change_blender_281](https://user-images.githubusercontent.com/479115/69425278-45be3880-0cf0-11ea-8e73-cbb0aa9e28b9.png)

![iris_change_blender_280](https://user-images.githubusercontent.com/479115/69425279-45be3880-0cf0-11ea-9205-7e9fd51716a9.png)

**Comment by animate1978** [comment]
Side note. While the Min Blender version is set now 2.81.16, MB-Lab will in fact run in Blender 2.80.75, I just tested it and MB-Lab will function as expected.

**Comment by animate1978** [comment]
Found the real root of the problem. It is in the shader nodes. When it's saved as a 2.80 file it's fine, until the file is saved in 2.81 where some of the nodes have changed.

![shadebug_test_02](https://user-images.githubusercontent.com/479115/69439176-069de080-0d0c-11ea-94c1-490af347cf74.png)

The above is the nodes in 2.81

![shadebug_test](https://user-images.githubusercontent.com/479115/69439174-06054a00-0d0c-11ea-8a47-6957e8bc98a1.png)

Same file but saved and opened in 2.80.

With the changes in 2.81 I was afraid this would happen. Again it's bad practice to open up files from latest versions in previous as I stated above but people do things that are unexpected. This could be one of them and now we know what happens.
